test/provider_devdax_memory.cpp

```diff
@@ -0,0 +1,289 @@
+                       UMF_RESULT_ERROR_INVALID_ARGUMENT, 0);
+}
+
+TEST_P(umfProviderTest, alloc_WRONG_SIZE) {
+    test_alloc_failure(provider.get(), -1, 0,
+                       UMF_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC,
+                       UMF_DEVDAX_RESULT_ERROR_ALLOC_FAILED);
+}
+
+// other positive tests
+
+TEST_P(umfProviderTest, get_min_page_size) {
+    size_t min_page_size;
+    umf_result_t umf_result = umfMemoryProviderGetMinPageSize(
+        provider.get(), nullptr, &min_page_size);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_LE(min_page_size, page_size);
+}
+
+TEST_P(umfProviderTest, get_recommended_page_size) {
+    size_t min_page_size;
+    umf_result_t umf_result = umfMemoryProviderGetMinPageSize(
+        provider.get(), nullptr, &min_page_size);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_LE(min_page_size, page_size);
+
+    size_t recommended_page_size;
+    umf_result = umfMemoryProviderGetRecommendedPageSize(
+        provider.get(), 0, &recommended_page_size);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_GE(recommended_page_size, min_page_size);
+}
+
+TEST_P(umfProviderTest, get_name) {
+    const char *name = umfMemoryProviderGetName(provider.get());
+    ASSERT_STREQ(name, "DEVDAX");
+}
+
+TEST_P(umfProviderTest, free_size_0_ptr_not_null) {
+    umf_result_t umf_result =
+        umfMemoryProviderFree(provider.get(), INVALID_PTR, 0);
+    ASSERT_EQ(umf_result, UMF_RESULT_ERROR_NOT_SUPPORTED);
+}
+
+TEST_P(umfProviderTest, free_NULL) {
+    umf_result_t umf_result = umfMemoryProviderFree(provider.get(), nullptr, 0);
+    ASSERT_EQ(umf_result, UMF_RESULT_ERROR_NOT_SUPPORTED);
+}
+
+// other negative tests
+
+TEST_P(umfProviderTest, free_INVALID_POINTER_SIZE_GT_0) {
+    umf_result_t umf_result =
+        umfMemoryProviderFree(provider.get(), INVALID_PTR, page_plus_64);
+    ASSERT_EQ(umf_result, UMF_RESULT_ERROR_NOT_SUPPORTED);
+}
+
+TEST_P(umfProviderTest, purge_lazy_INVALID_POINTER) {
+    umf_result_t umf_result =
+        umfMemoryProviderPurgeLazy(provider.get(), INVALID_PTR, 1);
+    ASSERT_EQ(umf_result, UMF_RESULT_ERROR_NOT_SUPPORTED);
+}
+
+TEST_P(umfProviderTest, purge_force_INVALID_POINTER) {
+    umf_result_t umf_result =
+        umfMemoryProviderPurgeForce(provider.get(), INVALID_PTR, 1);
+    ASSERT_EQ(umf_result, UMF_RESULT_ERROR_MEMORY_PROVIDER_SPECIFIC);
+
+    verify_last_native_error(provider.get(),
+                             UMF_DEVDAX_RESULT_ERROR_PURGE_FORCE_FAILED);
+}
+
+// negative tests
+
+TEST_F(test, create_wrong_path) {
+    umf_memory_provider_handle_t hProvider = nullptr;
+    const char *path = "/tmp/dax0.0";
+    auto wrong_params =
+        umfDevDaxMemoryProviderParamsDefault((char *)path, 4096);
+    auto ret = umfMemoryProviderCreate(umfDevDaxMemoryProviderOps(),
+                                       &wrong_params, &hProvider);
+    EXPECT_EQ(ret, UMF_RESULT_ERROR_INVALID_ARGUMENT);
+    EXPECT_EQ(hProvider, nullptr);
+}
+
```

Comment:
could you also pls add a test with path like this `/tmp/dev/dax0.0` so we make sure we only allow paths starting with `/dev/dax`, not just containing `/dev/dax`